core/src/main/java/org/apache/accumulo/core/iteratorsImpl/system/VisibilityFilter.java

```diff
@@ -64,7 +66,7 @@ public synchronized SortedKeyValueIterator<Key,Value> deepCopy(IteratorEnvironme
 
   @Override
   protected boolean accept(Key k, Value v) {
-    ByteSequence testVis = k.getColumnVisibilityData();
+    k.getColumnVisibilityData(testVis);
```

Comment:
There may be a bug in these change where `defaultVisibility` is modified because a previous call to this method could have done `testVis=defaultVisibility`.  So may end up modifying `defaultVisibility` when calling ` k.getColumnVisibilityData(testVis)` here.

The general discussion of mutable vs immutable on this PR and seeing this lead to opening the draft PR #4748.

Reply:
Really good catch here. Proposed solution added in 2012166